File: MicroAppFront/src/app/products/models/product.ts
import { Document } from './../../shared/documents/models/document';
import { ProductCategory } from './productCategory';

export interface Product {
  id: string;
  name: string;
  price: number;
  description: string;
  categoryId: number;
  productCategory?: ProductCategory;
  document?: Document;
  image?: any;
}

File: MicroApp/MicroApp.Api/Models/Products/Product.cs
using MicroApp.Api.Models.Documents;
using System;

namespace MicroApp.Api.Models.Products
{
    public class Product
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public Guid CategoryId { get; set; }
        public Document Document { get; set; }
    }
}
